qutip/solver/stochastic.py

```diff
@@ -626,6 +648,85 @@ def _integrate_one_traj(self, seed, tlist, result):
             result.add(t, self._restore_state(state, copy=False), noise)
         return seed, result
 
+    def run_from_experiment(
+        self, state, tlist, noise, *,
+        args=None, e_ops=(), measurement=False,
+    ):
+        """
+        Run a single trajectory from a given state and noise.
+
+        Parameters
+        ----------
+        state : Qobj
+            Initial state of the system.
+
+        tlist : array_like
+            List of times for which to evaluate the state. The tlist must
+            increase uniformly.
+
+        noise : array_like
+            Noise for each time step and each stochastic collapse operators.
+            For homodyne detection, ``noise[i, t_idx]`` is the Wiener
+            increments between ``tlist[t_idx]`` and ``tlist[t_idx+1]`` for the
+            i-th sc_ops.
+            For heterodyne detection, an extra dimension is added for the pair
+            of measurement: ``noise[i, j, t_idx]``with ``j`` in ``{0,1}``.
+
+        args : dict, optional
+            Arguments to pass to the Hamiltonian and collapse operators.
+
+        e_ops : list, optional
+            List of operators for which to evaluate expectation values.
+
+        measurement : bool, default : False
+            Whether the passed noise is the Wiener increments ``dW`` (gaussian
+            noise with standard derivation of dt**0.5), or the measurement::
+
+                noise = dW/dt * dW_factors
+                    + expect(sc_ops[i] + sc_ops[i].dag, state_t)
+
+            Note that the expectation value is usally computed at the start of
+            the step. Only available for limited integration methods.
```

Comment:
I found this sentence a bit unclear. Why "usually"?

```suggestion
            Note that this function expects the expectation values to be taken at the start of
            the time step, corresponding to the "start" setting for the "store_measurements" option.
            Only available for limited integration methods.
```